src/opencensus/proto/agent/metrics/v1/metrics_service.proto

```diff
@@ -42,6 +42,10 @@ message ExportMetricsServiceRequest {
 
   // A list of metrics that belong to the last received Node.
   repeated opencensus.proto.metrics.v1.Metric metrics = 2;
+
+  // The resource for the metrics in this message. If unset, the resource of the
+  // most recent node applies.
+  opencensus.proto.agent.common.v1.Resource resource = 3;
```

Comment:
@bogdandrutu this allows to overwrite the default resource for the batch of metrics in this request.

Here I've been wondering, whether it would make sense to remove the resource field from node, and just give this resource field here the same semantics as the node field, i.e. the most recent set value applies.
It removes one layer of nesting and gives both fields the same defaulting semantics, which seems to make semantics a bit simpler.

Just an alternative though – nothing fundamentally superior.

Reply:
Should `Resource` be part of `metrics` instead of just `metrics_service`?

Reply:
You mean out of the agent package and into the top level metrics package?
Given resource will not only apply to metrics but also spans (and potentially other signals in the future), it would rather need its own top-level package then.

I thought about that but then discarded the idea since it seems we can avoid attaching it to the signal-specific messages entirely.

Reply:
> Given resource will not only apply to metrics but also spans (and potentially other signals in the future), it would rather need its own top-level package then.

I agree. I'm just wondering if there are any other sources (other than OC-Agent, for example library exporters) that expect `resource` along with `metrics`. With the current approach, `resource` is only available to OC-Agent.

Reply:
Yes, you are right I think. Talked to Bogdan about this earlier, and giving signals an explicit resource, which may be empty in many cases and defaulted to what is provided in the agent protocol stream seems better.